src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -58,29 +58,44 @@ public static bool RunningOnRHEL6
         {
             get
             {
-                if (!RunningOnLinux || !File.Exists("/etc/redhat-release"))
+                if (!(detectedRHEL6 is null))
                 {
-                    return false;
+                    return (bool)detectedRHEL6;
                 }
 
+                DetectRHEL6();
+
+                return (bool)detectedRHEL6;
+            }
+        }
+
+        private static void DetectRHEL6()
```

Comment:
do we need to worry about thread safety here? This is one of the corners of C# I am not up to speed on which is why I am asking.